src/sidebar/components/test/thread-test.js

```diff
@@ -0,0 +1,284 @@
+import { mount } from 'enzyme';
+import { createElement } from 'preact';
+import { act } from 'preact/test-utils';
+
+import Thread from '../thread';
+import { $imports } from '../thread';
+
+import { checkAccessibility } from '../../../test-util/accessibility';
+import mockImportedComponents from '../../../test-util/mock-imported-components';
+
+// Utility functions to build nested threads
```

Comment:
I've got an eye on this; creating some reusable `Thread` fixtures would be handy.